File: cs50x/Problem Set 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int calculate_letters(string text);
int calculate_words(string text);
int calculate_sentences(string text);

int main(void)
{
    string text = get_string("Enter text: ");

    float letters = calculate_letters(text);
    float words = calculate_words(text);
    float sentences = calculate_sentences(text);

    float L = letters / words * 100;
    float S = sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", (int) round(index));
    }
}

int calculate_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int calculate_words(string text)
{
    int spaces = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            spaces++;
        }
    }
    return spaces + 1;
}

int calculate_sentences(string text)
{
    int punctuations = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            punctuations++;
        }
    }
    return punctuations;
}

File: cs50x/Problem Set 6/sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    text = get_string('Text: ')

    lenght = len(text)

    words_counter = 0
    sentences_counter = 0
    letters_counter = 0
    #provjerit zasto u istom redu nemos imat vise varijabli i svima im dodjelit istu int vrijednost recimo 0

    for char in text:
        if char == ' ':
            pass
            words_counter += 1

        elif char in '.!?':
            sentences_counter += 1

        elif char in 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ':
            letters_counter += 1

    words = words_counter + 1
    sentences = sentences_counter
    letters = letters_counter

    L = letters / words * 100
    S = sentences / words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)
    #print(index)

    if index < 1:
        print('Before Grade 1')

    elif index >= 16:
        print('Grade 16+')

    else:
        print('Grade', index)

if __name__ == '__main__':
    main()